addons/sale_timesheet/models/sale_order.py

```diff
@@ -24,6 +24,9 @@ class SaleOrder(models.Model):
         help='Select a non billable project on which tasks can be created.')
     project_ids = fields.Many2many('project.project', compute="_compute_project_ids", string='Projects', copy=False, groups="project.group_project_user", help="Projects used in this sales order.")
     timesheet_encode_uom_id = fields.Many2one('uom.uom', related='company_id.timesheet_encode_uom_id')
+    timesheet_encoded_in_hours = fields.Boolean(compute='_compute_timesheet_total', readonly=True)
```

Comment:
I think one field with the total duration computed in the encoded uom of the document's company will be enough, as we already have the related field "timesheet_encode_uom_id".
In the compute, you can supposed the UOM of timesheet is either its product_uom_id field, or the hardcoded hour uom (from data). Then convert all unit amount from timesheet into the encoded uom of SO company.

Reply:
done